liblwgeom/lwgeom_geos.c

```diff
@@ -1206,8 +1210,11 @@ lwgeom_sharedpaths(const LWGEOM* geom1, const LWGEOM* geom2)
 	return result;
 }
 
-LWGEOM*
-lwgeom_offsetcurve(const LWLINE* lwline, double size, int quadsegs, int joinStyle, double mitreLimit)
+static LWGEOM*
+lwgeom_offsetcurve_line(const LWLINE* lwline, double size, int quadsegs, int joinStyle, double mitreLimit);
+
+static LWGEOM*
+lwgeom_offsetcurve_line(const LWLINE* lwline, double size, int quadsegs, int joinStyle, double mitreLimit)
```

Comment:
`lwline_offsetcurve` and `lwmline_offsetcurve` would more closely follow existing conventions, IMO

Reply:
Sorry, maybe this comment was unclear. The convention would be to have a function `lwgeom_offsetcurve` that takes an argument `LWGEOM` and uses the `switch` statement to dispatch to `lwline_offsetcurve`, `lwmline_offsetcurve`, `lwcollection_offsetcurve`, etc. as needed. They type-specific variants can all be public. See for example `lwgeom_segmentize2d` in the header at line 1481.